experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1373 @@
+method. See there for further information.
+
+# Examples
+
+```jldoctests
+julia> S, variables = differential_polynomial_ring(QQ, 3, 4)
+(Differential polynomial ring in 3 elementary symbols over QQ, DifferentialPolyRingElem{QQFieldElem}[u1[0,0,0,0], u2[0,0,0,0], u3[0,0,0,0]])
+
+julia> S
+Differential polynomial ring in 3 elementary symbols u1, u2, u3
+with 4 commuting derivations
+  over Rational field
+
+julia> variables
+3-element Vector{DifferentialPolyRingElem{QQFieldElem}}:
+ u1[0,0,0,0]
+ u2[0,0,0,0]
+ u3[0,0,0,0]
+```
+"""
+function differential_polynomial_ring(R::Ring, elementary_symbols::Vector{Symbol}, ndiffs::Int;
+    partition_name::Symbol = :default,
+    index_ordering_name::Symbol = :default,
+    partition::Vector{Vector{Int}} = Vector{Int}[],
+    index_ordering_matrix::ZZMatrix = zero_matrix(ZZ, 0, 0),
+  )
+  dpr = DifferentialPolyRing{elem_type(typeof(R))}(R, elementary_symbols, ndiffs)
+  set_ranking!(dpr;
+      partition_name = partition_name,
+      index_ordering_name = index_ordering_name,
+      partition = partition,
+      index_ordering_matrix = index_ordering_matrix
+    )
+  return (dpr, deepcopy.(__add_new_jetvar!(dpr, map(i -> (i, fill(0, ndiffs)), 1:length(elementary_symbols)))))
+end
+
+##### Elements #####
+
+### Difference ###
+(dpr::DifferencePolyRing)() = DifferencePolyRingElem{elem_type(base_ring(dpr))}(dpr)
+
+(dpr::DifferencePolyRing)(upre::AbstractAlgebra.Generic.UniversalPolyRingElem) = DifferencePolyRingElem{elem_type(base_ring(dpr))}(dpr, upre)
+
+function (dpr::DifferencePolyRing{T})(a::DifferencePolyRingElem{T}) where {T}
+  @req parent(a) === dpr "Wrong parent"
+  return a
+end
+
+(dpr::DifferencePolyRing)(a::T) where {T<:RingElement} = dpr(dpr.upoly_ring(a))
+
+### Differential ###
+(dpr::DifferentialPolyRing)() = DifferentialPolyRingElem{elem_type(base_ring(dpr))}(dpr)
+
+(dpr::DifferentialPolyRing)(upre::AbstractAlgebra.Generic.UniversalPolyRingElem) = DifferentialPolyRingElem{elem_type(base_ring(dpr))}(dpr, upre)
+
+function (dpr::DifferentialPolyRing{T})(a::DifferentialPolyRingElem{T}) where {T}
+  @req parent(a) === dpr "Wrong parent"
+  return a
+end
+
+(dpr::DifferentialPolyRing)(a::T) where {T<:RingElement} = dpr(dpr.upoly_ring(a))
+
+###############################################################################
+#
+#  Basic ring functionality 
+#
+###############################################################################
+
+### Difference ###
+elem_type(::Type{DifferencePolyRing{T}}) where {T} = DifferencePolyRingElem{T}
+
+parent_type(::Type{DifferencePolyRingElem{T}}) where {T} = DifferencePolyRing{T}
+
+function Base.deepcopy_internal(dpre::DifferencePolyRingElem{T}, dict::IdDict) where {T}
+    # Avoid deepcopying the parent as it may refer back to it in one of its dictionaries 
+    pp = deepcopy_internal(data(dpre), dict)
+    return DifferencePolyRingElem{T}(parent(dpre), pp)
+end
+
+zero(dpr::DifferencePolyRing) = dpr(0)
+
+one(dpr::DifferencePolyRing) = dpr(1)
```

Comment:
```suggestion
one(dpr::DifferencePolyRing) = dpr(one(dpr.upoly_ring))
```
this is more efficient